pkg/controller/pools.go

```diff
@@ -92,6 +93,39 @@ func poolSSMatchesSpec(expectedStatefulSet, existingStatefulSet *appsv1.Stateful
 	return true, nil
 }
 
+// getSystemCfgFromStatefulSet gets the MinIO environment variables from a statefulset
+// set getServerSystemCfg at minio
+func (c *Controller) getSystemCfgFromStatefulSet(ctx context.Context, sts *appsv1.StatefulSet) (systemCfg map[string]string, err error) {
+	systemCfg = make(map[string]string)
+	for _, container := range sts.Spec.Template.Spec.Containers {
+		if container.Name == miniov2.MinIOServerName {
+			for _, e := range container.Env {
+				if strings.HasPrefix(e.Name, "MINIO_") {
+					switch {
+					case e.Value != "":
+						systemCfg[e.Name] = e.Value
+					case e.ValueFrom != nil && e.ValueFrom.SecretKeyRef != nil:
+						secret, err := c.kubeClientSet.CoreV1().Secrets(sts.Namespace).Get(ctx, e.ValueFrom.SecretKeyRef.Name, metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						systemCfg[e.Name] = string(secret.Data[e.ValueFrom.SecretKeyRef.Key])
```

Comment:
We may want to check if the key exists:
```suggestion
						value, ok := secret.Data[e.ValueFrom.SecretKeyRef.Key]
						if !ok {
							return nil, fmt.Errorf("key %q does not exist in secret %s/%s", e.ValueFrom.SecretKeyRef.Key, sts.Namespace, e.ValueFrom.SecretKeyRef.Name)
						}
						systemCfg[e.Name] = string(value)
```